### PR TITLE
Do not change "startVm" value when selecting Unattended installation

### PR DESCRIPTION
**Pull request by skobyda**
This was causing several in several bugs in Create VM dialog:
1. uncheck "Start VM" checkbox, select OS with aunattended installation,
   click "X" button, "Start VM" button is checked.
2. uncheck "Start VM" checkbox, check "Unattended installation", uncheck
   "Unattended installation", "Start VM" checkbox is checked.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=2033603

### CODE REVIEW

**Review by jelly — APPROVED** [review]

**Comment by martinpitt** [comment]
test/check-machines-create:800:85: W291 trailing whitespace

File: src/components/vm/hostdevs/hostDevAdd.jsx [hunk 1]
```diff
@@ -88,10 +88,10 @@ const DevRow = ({ idPrefix, type, selectableDevices, setSelectableDevices }) =>
...
             cells.push(getOptionalValue(bus, `${id}-bus`, _("Bus")));
         } else if (nodeDev.capability.type === "pci") {
-            let domain = nodeDev.capability.domain;
-            let bus = nodeDev.capability.bus;
-            let slot = nodeDev.capability.slot;
-            let func = nodeDev.capability.function;
+            let domain = Number(nodeDev.capability.domain);
+            let bus = Number(nodeDev.capability.bus);
+            let slot = Number(nodeDev.capability.slot);
+            let func = Number(nodeDev.capability.function);
```
Comment (martinpitt) [inline src/components/vm/hostdevs/hostDevAdd.jsx:94]: This is an unrelated change -- can you please move it to a separate commit?

File: test/check-machines-create [hunk 2]
```diff
@@ -798,9 +798,12 @@ class TestMachinesCreate(VirtualMachinesCase):
             b.wait_not_present("#navbar-oops")
 
             # click the 'X' button to clear the OS input and check there is no Ooops
+            # Also check start VM checkbox was not changed (https://bugzilla.redhat.com/show_bug.cgi?id=2033603)
+            b.set_checked("#start-vm", False)
             b.click("#os-select-group button[aria-label=\"Clear all\"]")
             b.wait_attr("#os-select-group input", "value", "")
             b.wait_not_present("#navbar-oops")
+            self.assertTrue(not b.get_checked("#start-vm"))
```
Comment (martinpitt) [inline test/check-machines-create:806]: assertFalse() and drop the `not`

File: src/components/create-vm-dialog/createVmDialog.jsx [hunk 3]
```diff
@@ -925,7 +927,7 @@ class CreateVmModal extends React.Component {
                 storageSize: convertToUnit(this.state.storageSize, this.state.storageSizeUnit, units.GiB),
                 storagePool: this.state.storagePool,
                 storageVolume: this.state.storageVolume,
-                startVm: this.state.startVm,
+                startVm: this.state.unattendedInstallation || this.state.startVm,
```
Comment (martinpitt) [inline src/components/create-vm-dialog/createVmDialog.jsx:930]: This is weird UI-wise -- it means you can uncheck the "immediately start" checkbox, but it will still auto-start the VM anyway, because it's unattended? If unattended install really doesn't work with a delayed start, then please add the reason as a comment here. But then the previous behaviour was actually right, and the checkbox *should* be auto-checked when you select unattended. But then it should also be disabled in that case.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! Please rebase this, so that packit tests can run.

File: src/components/vm/hostdevs/hostDevAdd.jsx (hunk 1 above)
Comment (skobyda) [reply src/components/vm/hostdevs/hostDevAdd.jsx:94]: Ahhh, how this got here, this should not even be in his PR

File: src/components/create-vm-dialog/createVmDialog.jsx (hunk 3 above)
Comment (skobyda) [reply src/components/create-vm-dialog/createVmDialog.jsx:930]: > This is weird UI-wise -- it means you can uncheck the "immediately start" checkbox, but it will still auto-start the VM anyway, because it's unattended? If unattended install really doesn't work with a delayed start, then please add the reason as a comment here. But then the previous behaviour was actually right, and the checkbox _should_ be auto-checked when you select unattended. But then it should also be disabled in that case.

No, that is exactly current behavior which this PR doesn't ammend, it was like that before this PR and will still be even with this PR. When unattended installation checkbox is checked, then the startVm checkbox is disabled and automatically shown as checked.

What I'm trying to change in this PR is that startVm value doesn't get overridden in two bugged cases:
1. uncheck "Start VM" checkbox, select OS with unattended installation, check "Unattended installation", uncheck "Unattended installation", "Start VM" checkbox is checked.
2. uncheck "Start VM" checkbox, select OS with unattended installation,  click "X" button, "Start VM" button is checked.
So to fix both cases, I changed
```
-  this.setState({ unattendedInstallation: value, startVm: true });
+ this.setState({ unattendedInstallation: value });
```
Because when you click on "X" button, it calls onValueChanged("os", null), which in turns call onValueChanged("unattendedInstallation", false), which calls the above mentioned code, which I think wasn't logical because what it basically did was that whenever state of "unattendedInstallation" changed, it just set startVm to true which was cases both above mentioned bugs.

File: src/components/vm/hostdevs/hostDevAdd.jsx (hunk 1 above)
Comment (skobyda) [reply src/components/vm/hostdevs/hostDevAdd.jsx]: fixed

File: test/check-machines-create (hunk 2 above)
Comment (skobyda) [reply test/check-machines-create]: fixed

**Comment by martinpitt** [comment]
[This](https://logs.cockpit-project.org/logs/pull-494-20220114-141226-ec0e0c74-fedora-35-firefox/log.html#29-2) is an interesting flake, given the [screenshot](https://logs.cockpit-project.org/logs/pull-494-20220114-141226-ec0e0c74-fedora-35-firefox/TestMachinesMigration-testCopyStorageMigration-fedora-35-127.0.0.2-2501-FAIL.png) -- VM migration fails due to an SSH key prompt and then an "input/output error". Either the former always happens and something went wrong in the networking (although that's all local), or the SSH prompt is what actually breaks the copy, as nothing answers it.

However, it is unrelated. I'll push a clean rebase, to also run the packit tests.

File: src/components/create-vm-dialog/createVmDialog.jsx (hunk 3 above)
Comment (martinpitt) [reply src/components/create-vm-dialog/createVmDialog.jsx:930]: Makes sense, thanks! I tested this now, and it works correctly.

**Review by martinpitt — APPROVED** [review]
Cheers!

**Comment by martinpitt** [comment]
Hmm, the delete race is [really](https://logs.cockpit-project.org/logs/pull-494-20220118-063123-a344679c-arch/log.html#65-1) [bad](https://logs.cockpit-project.org/logs/pull-494-20220118-063125-a344679c-rhel-8-5/log.html#67) today -- I already retried once, trying again.

**Comment by martinpitt** [comment]
No really, [it's *hard*](https://logs.cockpit-project.org/logs/pull-494-20220118-092456-a344679c-rhel-8-5/log.html)!
